packages/salesforcedx-vscode-core/src/util/configUtil.ts

```diff
@@ -22,19 +26,47 @@ export enum ConfigSource {
   None
 }
 
-// This class should be reworked or removed once the ConfigAggregator correctly checks
-// local as well as global configs. It's also worth noting that ConfigAggregator, according
-// to its docs checks local, global and environment and, for our purposes, environment may
-// not be viable.
+async function getConfigAggregator(): Promise<ConfigAggregator> {
+  const origCurrentWorkingDirectory = process.cwd();
+  const rootWorkspacePath = getRootWorkspacePath();
+  // Change the current working directory to the project path,
+  // so that ConfigAggregator reads the local project values
+  process.chdir(rootWorkspacePath);
+  const configAggregator = await ConfigAggregator.create();
+  // Change the current working directory back to what it was
+  // before returning
+  process.chdir(origCurrentWorkingDirectory);
+  return configAggregator;
+}
+
+// The SfdxConfigAggregator is used only to get configuration
+// values that correspond with old/deprecated config keys.
+// Currently, the key used for the custom templates
+// directory is the only usage, since it is documented for use
+// here: https://developer.salesforce.com/tools/vscode/en/user-guide/byotemplate#set-default-template-location
+async function getSfdxConfigAggregator(): Promise<ConfigAggregator> {
+  const origCurrentWorkingDirectory = process.cwd();
+  const rootWorkspacePath = getRootWorkspacePath();
+  // Change the current working directory to the project path,
+  // so that ConfigAggregator reads the local project values
+  process.chdir(rootWorkspacePath);
+  const configAggregator = await SfdxConfigAggregator.create();
+  // Change the current working directory back to what it was
+  // before returning
+  process.chdir(origCurrentWorkingDirectory);
+  return configAggregator;
+}
 
 export class ConfigUtil {
-  public static async getConfigSource(key: string): Promise<ConfigSource> {
-    let value = await ConfigUtil.getConfigValue(key, ConfigSource.Local);
-    if (!isNullOrUndefined(value)) {
+  public static async getConfigSource(
+    key: string
+  ): Promise<ConfigSource.Local | ConfigSource.Global | ConfigSource.None> {
+    const configAggregator = await getConfigAggregator();
+    const configSource = configAggregator.getLocation(key);
+    if (configSource === ConfigAggregator.Location.LOCAL) {
```

Comment:
nit: can a switch statement on configSource work here?  I always forget the deets of when switch does and doesn't work in javascript land.